### PR TITLE
Updated directory traversal, tests and readme

### PR DESCRIPTION
**Pull request by manatarms**
This is a big PR but it makes the tree traversal super nice and centralized. I also updated all the tests. See comments.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -55,7 +55,7 @@
   "dependencies": {
     "chalk": "^1.1.3",
     "commander": "^2.9.0",
-    "directory-tree": "^1.2.1",
+    "directory-tree": "^2.0.0",
```
Comment (manatarms) [inline package.json:58]: This is the biggest change. My PR for this plugin got merged so we can now use a regex for comparisons.

File: src/stats.js [hunk 2]
```diff
@@ -112,15 +112,16 @@ Object.keys(config).forEach(frameworkNumber => {
   ) {
     // Traverse the subkeys of the framework
 
-    Object.keys(config[frameworkNumber]).forEach(typeOfCode => {
-      // Do this only if a typeOfCode is a src or a test
-      if (typeOfCode === CONFIG_KEYS.SRC || typeOfCode === CONFIG_KEYS.TEST) {
+    Object.keys(config[frameworkNumber]).forEach(typeOfKey => {
```
Comment (manatarms) [inline src/stats.js:115]: `typeOfKey` is more correct than `typeOfCode`

File: src/stats.js [hunk 3]
```diff
@@ -189,19 +197,6 @@ Object.keys(config).forEach(frameworkNumber => {
         }, {});
         // For each subdirectory, get LOC
         subdirectoryTree.files.forEach(file => {
-          // Read the file and get back the contents
```
Comment (manatarms) [inline src/stats.js:192]: We don't need all these checks because it's handled by the traversal function

File: src/utils/helpers.js [hunk 4]
```diff
@@ -14,28 +14,45 @@ const COMMON_KEYS = CONSTANTS.COMMON_KEYS;
...
  */
-let getChildren = (parent, currentChildren) => {
-  //If it has children, it is a directory
-  if (parent.children) {
-    //Push it into our directory array
-    currentChildren.children.push({
-      [COMMON_KEYS.PATH]: parent.path,
-      [COMMON_KEYS.LENGTH]: parent.children.length
-    });
+let getChildren = (parent, currentChildren, includeOnlyRegExp) => {
```
Comment (tajo) [inline src/utils/helpers.js:20]: why not const?

**Review by tajo — APPROVED** [review]

File: src/utils/helpers.js (hunk 4 above)
Comment (manatarms) [reply src/utils/helpers.js:20]: It runs through this loop 4 times so it can't be a const because it'll be reassigned.

**Pull request merged**